sklearn_extra/robust/robust_weighted_estimator.py

```diff
@@ -504,7 +561,7 @@ def score(self, X, y=None):
         return self.base_estimator_.score(X, y)
 
     def decision_function(self, X):
```

Comment:
So this should only be defined for classifiers right? 

Then I think something like this should work to define it conditionally
```suggestion

    @if_delegate_has_method(delegate='base_estimator')
    def decision_function(self, X):
```
https://scikit-learn.org/stable/modules/generated/sklearn.utils.metaestimators.if_delegate_has_method.html